### PR TITLE
Add issue template and docs for iceberg proposals

### PR DESCRIPTION
**Pull request by danielcweeks**
This PR adds an issue template for Iceberg improvement proposals and adds docs to the site to cover the proposal process.

### CODE REVIEW

**Comment by danielcweeks** [comment]
@jbonofre I'm not sure why I can't add you as a reviewer.

File: .github/ISSUE_TEMPLATE/iceberg_proposal.yml
Comment (liurenjie1024) [inline .github/ISSUE_TEMPLATE/iceberg_proposal.yml:49]: Should be this `Security`?

File: site/docs/contribute.md
Comment (liurenjie1024) [inline site/docs/contribute.md:49]:
```suggestion
## Apache Iceberg Improvement Proposals
```

**Comment by ajantha-bhat** [comment]
> @jbonofre I'm not sure why I can't add you as a reviewer.

What I observed is only collaborators, committers can be added as reviewer.
https://github.com/apache/iceberg/blob/main/.asf.yaml

File: site/docs/contribute.md
Comment (liurenjie1024) [inline site/docs/contribute.md:59]:
```suggestion
1. A GitHub issue created using the `Apache Iceberg Improvement Proposal` template
```

**Review by liurenjie1024 — COMMENTED** [review]
Thanks for this, just added some minor nits.

**Comment by jbonofre** [comment]
@danielcweeks That's because my user is not in the `collaborators` list in `.asf.yml`. Let me check with asfinfra if we can have more than 10 entries in `collaborators`.

File: .github/ISSUE_TEMPLATE/iceberg_proposal.yml
Comment (Fokko) [reply .github/ISSUE_TEMPLATE/iceberg_proposal.yml:49]: I think this specifically refers to the Encryption spec: https://iceberg.apache.org/gcm-stream-spec/

File: .github/ISSUE_TEMPLATE/iceberg_proposal.yml
Comment (nastra) [inline .github/ISSUE_TEMPLATE/iceberg_proposal.yml:38]:
```suggestion
        Link to the proposal document. Google Docs is preferred format to allow for public 
```

File: site/docs/contribute.md
Comment (nastra) [inline site/docs/contribute.md:73]:
```suggestion
Current proposals are tracked in GitHub issues with the label [proposal](https://github.com/apache/iceberg/issues?q=is%3Aissue+is%3Aopen+label%3Aproposal+)
```

should this maybe just directly link to the proposals?

File: .github/ISSUE_TEMPLATE/iceberg_proposal.yml
Comment (nastra) [inline .github/ISSUE_TEMPLATE/iceberg_proposal.yml:27]: nit: link might need to be updated depending on how the section is going to be named

**Review by nastra — APPROVED** [review]

File: site/docs/contribute.md
Comment (danielcweeks) [reply site/docs/contribute.md:73]: I prefer the reference since it keeps the text cleaner and is more consistent with other areas of the doc.

File: .github/ISSUE_TEMPLATE/iceberg_proposal.yml
Comment (liurenjie1024) [inline .github/ISSUE_TEMPLATE/iceberg_proposal.yml:21]:
```suggestion
name: Apache Iceberg Improvement Proposal
```

Comment (danielcweeks) [reply .github/ISSUE_TEMPLATE/iceberg_proposal.yml:21]: This change would be inconsistent with the other templates (not everything needs the prefix)

File: .github/ISSUE_TEMPLATE/iceberg_proposal.yml
Comment (danielcweeks) [reply .github/ISSUE_TEMPLATE/iceberg_proposal.yml]: Updated

**Review by liurenjie1024 — APPROVED** [review]

File: site/docs/contribute.md
Comment (jbonofre) [inline site/docs/contribute.md:54]: Maybe also mention "additional features".

Comment (danielcweeks) [reply site/docs/contribute.md:54]: added

File: site/docs/contribute.md
Comment (danielcweeks) [reply site/docs/contribute.md:77]: For spec/major changes I don't think we want to leave that up to lazy consensus.  We should try to achieve consensus before a vote, but the vote is really what codifies the change and removes any ambiguity.

Comment (jbonofre) [reply site/docs/contribute.md:77]: Yeah, I agree for spec/major change. My question was more for proposed new features. But yeah, better to have a vote. I'm just a little bit concerned by potential veto vote. Imagine if we have only +1 and just one -1 from a PMC member, the change/addition can be vetoed. But you are right, we should work on consensus before the vote :)

Comment (jbonofre) [reply site/docs/contribute.md:77]: Thanks !

**Review by Fokko — APPROVED** [review]
Looks great!

**Review by stevenzwu — APPROVED** [review]

**Review by RussellSpitzer — APPROVED** [review]
LGTM

**Review by rdblue — APPROVED** [review]

**Pull request merged**
